flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/dialect/JDBCDialect.java

```diff
@@ -115,6 +115,20 @@ default String getDeleteStatement(String tableName, String[] conditionFields) {
 		return "DELETE FROM " + quoteIdentifier(tableName) + " WHERE " + conditionClause;
 	}
 
+	/**
+	 * Get select fields statement by `is not distinct from` condition fields. Default use SELECT.
+	 */
+	default String getSelectNotDistinctFromStatement(String tableName, String[] selectFields, String[] conditionFields) {
```

Comment:
I would re-design the method names a bit. 

- Add a new method `getSelectStatement(String tableName, String[] selectFields)` which doesn't contain `WHERE` clause.
- Rename `getSelectFromStatement` to `getFilterStatement(String tableName, String[] selectFields, String[] conditionFields)` and forces `conditionFields` shouldn't be empty.
- Rename `getSelectNotDistinctFromStatement` to `getNullSafeFilterStatement(...)`, because the default implementation is not `is not distinct from` but is a null safe comparison different from `getFilterStatement`. We should also outline this in the method javadoc.